backend/src/test/java/com/sim_backend/state/StateTest.Java

```diff
@@ -0,0 +1,94 @@
+package com.sim_backend.state;
+
+import static org.mockito.Mockito.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+
+public class StateTest {
+
+  private SimulatorStateMachine testStateMachine;
+
+  @BeforeEach
+  void setUp() {
+    testStateMachine = new SimulatorStateMachine();
+  }
+
+  @Test
+  // Initialize test
+  void testInit() throws Exception {
+    try {
```

Comment:
This whole test function can be replaced with
```
    assertEquals(SimulatorState.PowerOff, testStateMachine.getCurrentState(),
    "StateMachine should initialize to PowerOff");
```

Have `import static org.junit.jupiter.api.Assertions.*;` to have access to the  functions.

The try/catch method used here is hard to read and will not report test failures correctly. Refactor all the tests you wrote (including IndicatorTest.java). 

If you want to discuss how to write tests or need further clarification, just let me know and I would be happy to assist.

Reply:
I just personally like try, catch, if
fixed by using assertEquals